### PR TITLE
SAM-2291 Changes to generate a column for each option

### PR DESCRIPTION
**Pull request by jonespm**
Created this from the patch mentioned on SAM-2291, needs review.

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Could we get descriptions in English please. Whilst we support multiple languages, its a waste of time for reviewers to translate descriptions. That said Google translate tells me this is 'Changes to generate a column for each option'.

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java [hunk 1]
```diff
@@ -2253,6 +2253,8 @@ public List getExportResponsesData(String publishedAssessmentId, boolean anonymo
 				  boolean addRationale = false;
 				  boolean addResponseComment = false;
 
+				  boolean matrixChoices = false;
+				  TreeMap responsesMap = new TreeMap();
```
Comment (steveswinsburg) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java]: I dont think a TreeMap is warranted here, unnecessary overhead. LinkedHashmap would suffice if order is required.

**Comment by jonespm** [comment]
I can't change the original branch, so if we're (I'm) going to need to make changes to it rather than @mateullas I'd probably have to close this and start another PR.

**Comment by mateullas** [comment]
First of all, sorry for the annoyances about the  commit comment, I wrongly supposed it was private.

We have revised the code and check what happens if we use LinkedHashMap instead of TreeMap. 
The result is wrong. When the responsesMap is iterated to feed the ResponseList, if it is a TreeMap returns the correct order, but in the case of the LinkedHashMap the order is in which the entries were put into the map, and the order is not correct in general.

@jonespm What do you need to change? Is it the TreeMap or something else? 
If it something different of TreeMap I could do the changes myself in the branch

**Comment by jonespm** [comment]
@mateullas @steveswinsburg recommended LinkedHashmap

**Comment by sinmsinm** [comment]
In this particular case LinkedHashMap will not work. The iterator goes over
elements in the order that they were put in. In this case it's needed to
iterate elements by ids order, so changing Tree for Linked would write
wrong results in the file cells.

El Dv, 13 Mar., 2015 19.18 Code Hugger (Matthew Jones) <
notifications@github.com> va escriure:

**Pull request merged**